app/scripts/service/authorized-backend.service.js

```diff
@@ -69,8 +69,11 @@ define([
       //console.log(this.getTokenValidityMessage());
       var owner = this;
 
-      owner.getHttpPromise(httpConfigObject).then(function (response) {
-        thenFunction(response);
+      return owner.getHttpPromise(httpConfigObject).then(function (response) {
+        if (thenFunction) {
```

Comment:
left thenFunction integration intact for now so as not to break the current integrations with this call